int32 voltage
float64 percentage
---